pkg/util/httpproxy/upgradeawareproxy.go

```diff
@@ -172,6 +188,7 @@ func (p *UpgradeAwareSingleHostReverseProxy) serveUpgrade(w http.ResponseWriter,
 	done := make(chan struct{}, 2)
 
 	go func() {
+		// TODO: strip any WWW-Authenticate challenge headers if the initial response is a 401
```

Comment:
The only way I think you could do that is if you call http.ReadResponse(backendConn, req), check for 401, and if it's not 401, write the response to requestHijackedConn and then start the io.Copy